### PR TITLE
Added initial support for getting paths from project.

### PR DESCRIPTION
**Pull request by scott-wilson**
Hey, this is still a bit of a work in progress. I wanted to start the pull request to start a dialog on supporting getting paths from the project (Issue #43). If you have an idea on the TODOs below, or want to go with a different direction to getting the current path, feel free to let me know!

Current TODOs:
  - The useCurrentProject is always true. Need to figure out what I'm doing wrong.
  - I am assuming that the first path is the website path. I should probably figure out a way to guess the right path.

### CODE REVIEW

**Comment by zhuochun** [comment]
Thanks for the work done!

My thought was when `config.get("siteLocalDir")` returns the same as `config.getDefault("siteLocalDir)`, it fallbacks to use project paths. So another config `useCurrentProject` is not needed.

Something like this in config.coffee get:

``` coffee
get: (key) ->
  value = @getProject(key) || @getUser(key) || @getEngine(key) || @getDefault(key)

  if key == "siteLocalDir" && value == @getDefault("siteLocalDir')
    # fallback to use current project path
  else
    value
```

**Comment by scott-wilson** [comment]
Can do! I'll try to get another pull request either today or tomorrow. Then, I'll see if there's a smart way that I can try to figure out the actual root path of the project. I'm thinking of looping through the projects and try to find the _config.yml or something along the lines of that.

**Comment by zhuochun** [comment]
Great, thanks! I think use the `atom.project` would be sufficient for most general cases now.

File: lib/utils.coffee [hunk 1]
```diff
@@ -22,6 +24,15 @@ getPackagePath = (segments...) ->
   segments.unshift(atom.packages.resolvePackagePath("markdown-writer"))
   path.join.apply(null, segments)
 
+getRootPath = ->
+  paths = atom.project.getPaths()
+
+  if !config.get("siteLocalDir")
+    return paths[0]
+  else
+    return config.get("siteLocalDir")
```
Comment (zhuochun) [inline lib/utils.coffee]: can be shorten to `config.get("siteLocalDir") || paths[0]`

File: lib/utils.coffee [hunk 2]
```diff
@@ -22,6 +24,15 @@ getPackagePath = (segments...) ->
   segments.unshift(atom.packages.resolvePackagePath("markdown-writer"))
   path.join.apply(null, segments)
 
+getRootPath = ->
+  paths = atom.project.getPaths()
```
Comment (zhuochun) [inline lib/utils.coffee]: `paths` could be empty array when Atom opens without a project.

File: lib/utils.coffee [hunk 3]
```diff
@@ -22,6 +24,15 @@ getPackagePath = (segments...) ->
...
+
+  if !config.get("siteLocalDir")
+    return paths[0]
```
Comment (zhuochun) [inline lib/utils.coffee]: `paths` could be empty array when Atom opened without a current project.

File: spec/utils-spec.coffee [hunk 4]
```diff
@@ -26,7 +26,7 @@ describe "utils", ->
 
     it "get the path to package file", ->
       root = atom.packages.resolvePackagePath("markdown-writer")
-      expect(utils.getPackagePath("CHEATSHEET.md")).toEqual("#{root}/CHEATSHEET.md")
+      expect(utils.getPackagePath("CHEATSHEET.md")).toEqual("#{root}CHEATSHEET.md")
```
Comment (zhuochun) [inline spec/utils-spec.coffee]: why `/` is removed?

**Comment by zhuochun** [comment]
Can you squash the commits afterwards so I easier for me to merge. Thanks.

**Comment by scott-wilson** [comment]
Hey,

I received your notes in my email, but the notes are not in the change list. I'll address them here:
- I'm not too crazy with shortening if statements to one line, since I think they make things harder to read and understand.
- I couldn't think of any good default for the case of an empty project array, so I defaulted to the global project root. If you have any suggestions for this, I'm all ears.
- I swapped the cheatsheet test's toEqual to use the path.join. This should be a lot better, since the path may contain an / at the end.
- I was planning on doing a squash once there's no more issues with the code. Unless you want me to do it sooner, then I have no problem with that.

File: lib/utils.coffee [hunk 5]
```diff
@@ -22,6 +24,21 @@ getPackagePath = (segments...) ->
   segments.unshift(atom.packages.resolvePackagePath("markdown-writer"))
   path.join.apply(null, segments)
 
+getRootPath = ->
+  paths = atom.project.getPaths()
+
+  if paths
+    proejctPath = paths[0]
```
Comment (zhuochun) [inline lib/utils.coffee]: typo: `projectPath`

**Comment by zhuochun** [comment]
Sorry. I deleted some comments without known it is already sent.

LGTM. Please fix the typo and squash. Thanks.

**Comment by scott-wilson** [comment]
Typo fixed, and commit squashed! Let me know if there's anything else that needs looking at.

**Comment by scott-wilson** [comment]
Argh. I broke something with the squash. One sec.

**Comment by zhuochun** [comment]
The `useCurrentProject` config commit seems not squashed?

**Comment by scott-wilson** [comment]
Thanks for catching that.

**Pull request merged**